src/proxy_configuration.js

```diff
@@ -188,7 +195,7 @@ export class ProxyConfiguration {
      * @returns {Promise<void>}
      */
     async initialize() {
-        if (!this.proxyUrls) {
+        if (!this.proxyUrls || this.newUrlFunction) {
```

Comment:
`!this.proxyUrls || !this.newUrlFunction` perhaps?

Reply:
This could use `if (this.usesApifyProxy) { }` as mentioned below.

Reply:
Agreed. Looks better!